src/components/ModalWithForm.js

```diff
@@ -17,6 +19,16 @@ export default class ModalWithForm extends Modal {
     return this._formValues;
   }
 
+  getFormName() {
+    return this._form.getAttribute("name");
+  }
+
+  renderLoading(isLoading, loadingText = "Saving...") {
+    isLoading
+      ? (this._saveButton.textContent = loadingText)
+      : (this._saveButton.textContent = this._buttonText);
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._form.addEventListener("submit", (e) => {
```

Comment:
```js

      this._handleFormSubmit(this._getInputValues());
      this.close();
```


* Please, delete `this.close();` from line 37 in `ModalWithForm` because you need to close only in `then`